compiler/src/main/scala/org/scalajs/core/compiler/PrepJSExports.scala

```diff
@@ -491,65 +435,8 @@ trait PrepJSExports { this: PrepJSInterop =>
 
       actualExportInfos
     }
-  }
-
-  private def inheritedExportsOf(sym: Symbol): List[ExportInfo] = {
-    // The symbol from which we (potentially) inherit exports. It also
-    // gives the exports their name
-    val trgSym = {
-      if (sym.isModuleClass || (sym.isClass && isJSAny(sym))) {
-        sym
-      } else if (sym.isConstructor && sym.isPublic && !isJSAny(sym.owner) &&
-          sym.owner.isConcreteClass && !sym.owner.isModuleClass) {
-        sym.owner
-      } else {
-        NoSymbol
-      }
-    }
 
-    if (trgSym == NoSymbol) {
-      Nil
-    } else {
-      val trgAnnot =
-        if (sym.isModuleClass) JSExportDescendentObjectsAnnotation
-        else JSExportDescendentClassesAnnotation
-
-      val forcingSymInfos = for {
-        forcingSym <- trgSym.ancestors
-        annot      <- forcingSym.annotations
-        if annot.symbol == trgAnnot
-      } yield {
-        val ignoreInvalid = annot.constantAtIndex(0).fold(false)(_.booleanValue)
-        (forcingSym, ignoreInvalid)
-      }
-
-      // The dominating forcing symbol, is the first that does not ignore
-      // or the first otherwise
-      val forcingSymInfo =
-        forcingSymInfos.find(!_._2).orElse(forcingSymInfos.headOption)
-
-      val name = decodedFullName(trgSym)
-      val nameValid = !name.contains("__")
-
-      val optExport = for {
-        (forcingSym, ignoreInvalid) <- forcingSymInfo
-        if nameValid || !ignoreInvalid
-      } yield {
-        // Enfore no __ in name
-        if (!nameValid) {
-          // Get all annotation positions for error message
-          reporter.error(sym.pos,
-              s"${trgSym.name} may not have a double underscore (`__`) in " +
-              "its fully qualified name, since it is forced to be exported by " +
-              s"a @${trgAnnot.name} on $forcingSym")
-        }
-
-        ExportInfo(name, sym.pos, isNamed = false, ExportDestination.Normal,
-            ignoreInvalid)
-      }
-
-      optExport.toList
-    }
+    filteredExports.distinct
```

Comment:
Maybe we should warn or even error out in case of a duplicate export. Now that there are no inherited exports, duplicate exports would have to be written out, and that would be a bit suspicious.